zebedee-cms/src/test/java/com/github/onsdigital/zebedee/data/processing/DataPublicationDetailsTestBaseFixture.java

```diff
@@ -41,9 +43,15 @@
     CollectionWriter collectionWriter;
     DataBuilder dataBuilder;
     DataPagesGenerator generator;
+    SecretKey secretKey;
 
 
     public void setUp() throws Exception {
+        setUpPermissionsServiceMockForLegacyTests(zebedee, builder.publisher1);
+
+        secretKey = Keys.newSecretKey();
+        setUpKeyringMockForLegacyTests(zebedee, builder.publisher1, secretKey);
```

Comment:
This block appears common across the test base classes. Consider if it can be moved to the ZebedeeTestBaseFixture class.

Reply:
The block is calling some common set up in the base test class - I decided not to put this into the base set up as it would apply too all tests extending it - some tests uses different users/sessions/permissions depending on what its testing so I only wanted it to apply to the test cases that need it.

Plus its so complicated I'd rather never touch it again - it took much swearing to get it working 😂